src/Man.Dapr.Sidekick/Process/ManagedProcess.cs

```diff
@@ -31,12 +31,14 @@ public ManagedProcess()
         /// Starts a system process for a Dapr process binary.
         /// </summary>
         /// <param name="filename">The full path to the system process.</param>
+        /// <param name="workingDirectory">The working directory used by the process.</param>
         /// <param name="arguments">The optional command-line process arguments.</param>
         /// <param name="logger">An optional <see cref="IDaprLogger"/> instance for receiving stdout log messages from the process.</param>
         /// <param name="configureEnvironmentVariables">An optional action for configuring any environment variables for the process.</param>
         /// <param name="cancellationToken">A <see cref="DaprCancellationToken"/> for aborting the process startup operation.</param>
         public void Start(
             string filename,
+            string workingDirectory = null,
```

Comment:
This is a public class so inserting the optional `workingDirectory` parameter here will break binary interface compatibility with any existing applications that may have chosen to use this class directly. Suggest creating a new overload instead that adds `workingDirectory` to the end of the parameter list. Alternatively we could create a `ManagedProcessStartOptions` or `ManagedProcessStartSettings` class that wraps up all the parameters and add a `Start(ManagedProcessStartOptions options)` method signature so it would not change in the future as more options were added, thereby preserving compatibility. Which with hindsight is what we probably should have done in the first place.

Reply:
that is a good point, an overload with options makes sense, ill take care of it